### PR TITLE
add initial support for csky-unknown-linux-gnuabiv2

### PR DESCRIPTION
**Pull request by Dirreke**
Add libc-files to allow rust to be used on the csky/linux platform with a PR https://github.com/rust-lang/rust/pull/113658 for rust

Signed-off-by:  [@Dirreke](https://github.com/Dirreke) [mingyang_ge@163.com](mingyang_ge@163.com)

### CODE REVIEW

**Comment by Dirreke** [comment]
I know the code to support Linux/csky isn't yet in rust itself; https://github.com/rust-lang/rust/pull/113658 is in process. However, I need this crate merge first, and then https://github.com/rust-lang/rust/pull/113658 ensues because rust depends on libc by using `libc::c_char` 、`libc::c_int` ...
Thanks!

File: src/unix/linux_like/linux/gnu/b32/csky/align.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+s_no_extra_traits! {
+    #[allow(missing_debug_implementations)]
+    #[repr(align(2))]
+    pub struct max_align_t {
+        priv_: [i8; 20]
```
Comment (bjorn3) [inline src/unix/linux_like/linux/gnu/b32/csky/align.rs:5]: Is the size supposed to be significantly larger than the alignment or was it meant to be 2 too?

Comment (Dirreke) [reply src/unix/linux_like/linux/gnu/b32/csky/align.rs]: I will fix it, It's my ignorance.

**Comment by Dirreke** [comment]
@JohnTitor  May I ask if you have time to review this PR? Thanks

**Comment by JohnTitor** [comment]
Could you rebase and squash commits into one?

**Comment by Dirreke** [comment]
Done! Thanks

**Comment by Dirreke** [comment]
@JohnTitor  Could you merge it? Thanks!

**Pull request merged**